From db6e92352c5c6ec4578a88f883bce16c2a8570c2 Mon Sep 17 00:00:00 2001
From: jinboson <jinbo@loongson.cn>
Date: Mon, 9 Mar 2026 23:37:58 +0800
Subject: [PATCH] Fix test on loong64 (#4654)

After updating highway to 1.3.0 or master, AcStrategyDownsampleGroup/AcStrategyDownsample.Test/LSX_24 will fail.
Because the HWY_LSX version of MulAdd (implemented by __lsx_vfmadd_s(a,b,c)) starts producing different results
compared to the HWY_SCALAR version of MulAdd when the float point accuracy >= 1e-8. However, this is not saying
that the implemenation of HWY_LSX version of MulAdd is wrong, it's a hardware related implementation. So, I just
made this change on the test.

Co-authored-by: Eugene Kliuchnikov <eustas@google.com>
---
 lib/jxl/ac_strategy_test.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/jxl/ac_strategy_test.cc b/lib/jxl/ac_strategy_test.cc
index ed2d446f..ee0425bc 100644
--- a/lib/jxl/ac_strategy_test.cc
+++ b/lib/jxl/ac_strategy_test.cc
@@ -206,7 +206,7 @@ class AcStrategyDownsample : public ::hwy::TestWithParamTargetAndT<int> {
             sum /= 64;
             ASSERT_NEAR(
                 sum, idct_acs_downsampled[dy * 8 * acs.covered_blocks_x() + dx],
-                1e-6)
+                2e-6)
                 << " acs " << static_cast<int>(type);
           }
         }
-- 
2.56.0

